main.cpp

```diff
@@ -247,8 +247,8 @@ void clearResources(unsigned char *imageMat, float *boxArray)
 
 int main(int argc, char *argv[])
 {
-    char *filename = new char[0];
-    char *algorithm = new char[0];
+    char *filename = nullptr;
+    char *algorithm = nullptr;
```

Comment:
I'm not a huge C++ person, so this was a bit confusing. What was the point of doing `new char[0]` here?